core/domain/topic_domain.py

```diff
@@ -1137,6 +1146,33 @@ def create_default_topic(
             constants.DEFAULT_LANGUAGE_CODE, 0,
             feconf.CURRENT_STORY_REFERENCE_SCHEMA_VERSION, '', False, '')
 
+    @classmethod
+    def _convert_subtopic_v3_dict_to_v4_dict(cls, topic_id, subtopic_dict):
+        """Converts old Subtopic schema to the modern v4 schema. v4 schema
+        introduces the thumbnail_size_in_bytes field
+
+        Args:
+            topic_id: str. The id of the topic to which the subtopic is linked
+                to.
+            subtopic_dict: dict. A dict used to initialize a Subtopic domain
+                object.
+
+        Returns:
+            dict. The converted subtopic_dict.
+        """
+        file_system_class = fs_services.get_entity_file_system_class()
+        fs = fs_domain.AbstractFileSystem(file_system_class(
+            feconf.ENTITY_TYPE_TOPIC, topic_id))
+        filename_prefix = 'thumbnail'
```

Comment:
Make this generic, as we discussed in the other PR.

Reply:
Done, for now -- I will be making the change here as well.
This is being handled in [PR 13175](https://github.com/oppia/oppia/pull/13159#pullrequestreview-698846361). Once, that gets merged I will resolve the conflicts and merge this.